app/app.go

```diff
@@ -75,7 +81,16 @@ func addCachableHandler(e *gin.Engine, method string, endpoint string, generator
 // / This function will act as the handler for
 // / the home page
 func homeHandler(c *gin.Context, settings common.AppSettings, db database.Database) ([]byte, error) {
-	posts, err := db.GetPosts()
+	pageNumQuery := c.Param("num")
```

Comment:
a few issues with these additions. Firstly, this is user data, so this can be a negative number or empty, hence the `Atoi(pageNumQuery)` will fail.

I would personally do something like

```go
pageNum := 0
if pageNumQuery := c.Param("num"); pageNumQuery != "" {
num, err := strconv.Atoi(pageNumQuery)
if err == nil {
	pageNum = num
} else {
	log.Error().Msgf("invalid page number: %d", num)
}
}
```